agents/agents-core/src/commonMain/kotlin/ai/koog/agents/core/feature/model/events/agentEvents.kt

```diff
@@ -10,16 +10,34 @@ import kotlinx.serialization.Serializable
  * This event provides details about the agent's strategy, making it useful for
  * monitoring, debugging, and tracking the lifecycle of AI agents within the system.
  *
+ * @property id A unique identifier for the group of events associated with the agent execution event.
+ * @property parentId The unique identifier of the parent event, if applicable.
  * @property agentId The unique identifier of the AI agent;
  * @property runId The unique identifier of the AI agen run;
  * @property timestamp The timestamp of the event, in milliseconds since the Unix epoch.
  */
 @Serializable
 public data class AgentStartingEvent(
+    override val id: String,
+    override val parentId: String?,
     val agentId: String,
     val runId: String,
     override val timestamp: Long = Clock.System.now().toEpochMilliseconds(),
-) : DefinedFeatureEvent()
+) : DefinedFeatureEvent() {
+
+    /**
+     * @deprecated. Creates an instance of [AgentStartingEvent].
+     * Note! Do not relay on [id] and [parentId] parameters with this constructor.
+     */
+    @Deprecated(
```

Comment:
Here and in other similar places: I don't think it's actively used by anyone, let's just delete it to avoid unnecessary deprecations? Or is it important for Debugger functionality?